engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -2526,7 +2526,7 @@ private void orchestrateMigrateWithStorage(final String vmUuid, final long srcHo
                     final String zoneName = _dcDao.findById(vm.getDataCenterId()).getName();
                     boolean isWindows = _guestOSCategoryDao.findById(_guestOSDao.findById(vm.getGuestOSId()).getCategoryId()).getName().equalsIgnoreCase("Windows");
 
-                    vmData = _networkModel.generateVmData(userVm.getUserData(), serviceOffering, zoneName, vm.getInstanceName(), vm.getId(),
+                    vmData = _networkModel.generateVmData(userVm.getUserData(), serviceOffering, zoneName, vm.getInstanceName(), vm.getHostName(), vm.getId(),
```

Comment:
@mrunalinikankariya you want to send the name set by the user for the given VM, right? They are the parameters "displayname" and "name" that the user can enter when deploying a VM. 

Even though this "hostName" contains that parameter "name", I get the feeling that it is not the right one to use. It looks like an inconsistent sent of value that can and should be fixed in the future. If you look into the VM's POJO, we have hostId and hostName; for sanity, I would consider them the same thing. Therefore, hostId is the id of the host where the VM is running, and the hostName is the name of the host where the VM is running.

Therefore, I believe you should use "displayName" or "name" attributes instead of the "hostName".

Reply:
There is no displayname parameter in VirtualMachine.java. Also hostName refers to name parameter only

Reply:
Of course there is. I am not talking about the object you create we the call `generateVmData`. I am talking about the object from which you are retrieving data `vm`. That object is an instance of `UserVmVO`, which extends `VMInstanceVO`.